test/python/algorithms/evolvers/variational/calculators/test_evolution_grad_calculator.py

```diff
@@ -0,0 +1,215 @@
+                    (-0.6625999999999996 - 2e-18j),
+                    (0.43469999999999953 + 3e-18j),
+                    0j,
+                    (0.6625999999999996 - 9e-18j),
+                    (0.43469999999999975 + 1.4e-17j),
+                    (-2.5e-17 - 7e-18j),
+                    (-0.6625999999999992 + 3e-18j),
+                    (0.13419999999999993 + 9e-18j),
+                    (-0.13419999999999993 + 1.1e-17j),
+                    (-0.34349999999999975 - 9e-18j),
+                    (0.43469999999999953 - 1.1e-17j),
+                ],
+            ],
+        ),
+        (
+            Z - 1j * Y,
+            [
+                [
+                    (-0.38617868191914206 + 0.09029822949045127j),
+                    (-0.014055349300198292 - 0.19262478560541474j),
+                    (-0.06385049040183727 - 0.6510746098262302j),
+                    (0.13620629212619334 - 0.3879436778936384j),
+                    (-0.151807433390436 - 0.12595729749408652j),
+                    (-0.23783936538770684 - 0.3297422949956381j),
+                    (0.00240605468764643 - 0.5040008514312755j),
+                    (0.09977051760912456 - 0.33348367536103796j),
+                    (0.403577215950806 - 0.17322660907305487j),
+                    (0.010453846462186688 - 0.370182004647962j),
+                    (-0.04578581127401051 - 0.40796395449305006j),
+                    (0.04578581127401063 - 0.05062557967159678j),
+                ],
+                [
+                    (0.43469999999999975 + 3.6e-17j),
+                    (-2.1e-16 - 0.6625999999999996j),
+                    (2.25e-16 + 0.4346999999999995j),
+                    (0.6625999999999991 + 0j),
+                    (1.05e-16 + 0.6625999999999996j),
+                    (-3.3e-17 + 0.43469999999999975j),
+                    (-0.3434999999999998 - 2.5e-17j),
+                    (-1.58e-16 - 0.6625999999999992j),
+                    (1.02e-16 + 0.13419999999999993j),
+                    (-1.2e-16 - 0.13419999999999999j),
+                    (-4.2e-17 - 0.34349999999999975j),
+                    (-2.9e-17 + 0.4346999999999995j),
+                ],
+            ],
+        ),
+    )
+    @unpack  # TODO verify if values correct
+    def test_calculate_bases(self, basis, correct_values):
+        """Test calculating evolution gradient with non-default bases."""
+        observable = SummedOp(
+            [
+                0.2252 * (I ^ I),
+                0.5716 * (Z ^ Z),
+                0.3435 * (I ^ Z),
+                -0.4347 * (Z ^ I),
+                0.091 * (Y ^ Y),
+                0.091 * (X ^ X),
+            ]
+        ).reduce()
+
+        d = 2
+        ansatz = EfficientSU2(observable.num_qubits, reps=d)
+
+        # Define a set of initial parameters
+        parameters = ansatz.ordered_parameters
+        grad_method = "lin_comb"
+        evolution_grad = calculate(observable, ansatz, parameters, grad_method, basis)
```

Comment:
```suggestion
        grad_method = "lin_comb"
        evolution_grad = calculate(observable, ansatz, ansatz.parameters, grad_method, basis)
```

Reply:
Turns out it doesn't work.
```
"The linear combination gradient does only support the "
qiskit.opflow.exceptions.OpflowError: 'The linear combination gradient does only support the computation of 1st gradients and 2nd order gradients.'
```

Reply:
Hmmm interesting, maybe it's because `ansatz.parameters` is a `ParameterView` and opflow checks concretely for a list? If that's the case using `list(ansatz.parameters)` should do the trick.

Reply:
Correct, I replaced everything now.